src/main/java/novamachina/exnihilosequentia/common/init/ExNihiloBlocks.java

```diff
@@ -75,10 +74,31 @@
             .register(Blocks.CRUCIBLE_FIRED, FiredCrucibleBlock::new);
     public static final RegistryObject<BaseBlock> CRUCIBLE_WOOD = BLOCKS
             .register(Blocks.CRUCIBLE_WOOD, WoodCrucibleBlock::new);
-    public static final RegistryObject<BaseBlock> BARREL_WOOD = BLOCKS
-            .register(Blocks.BARREL_WOOD, () -> new BlockBarrel(new BlockBuilder().harvestLevel(ToolType.AXE, 0)
+    //TODO Able to Simplify???
+    public static final RegistryObject<BaseBlock> BARREL_ACACIA = BLOCKS
+            .register(Blocks.BARREL_ACACIA, () -> new BlockBarrel(new BlockBuilder().harvestLevel(ToolType.AXE, 0)
                     .properties(AbstractBlock.Properties.create(Material.WOOD).hardnessAndResistance(0.75F).sound(SoundType.WOOD))
-                    .tileEntitySupplier(WoodBarrelTile::new)));
+                    .tileEntitySupplier(AcaciaBarrelTile::new)));
+    public static final RegistryObject<BaseBlock> BARREL_BIRCH = BLOCKS
+            .register(Blocks.BARREL_BIRCH, () -> new BlockBarrel(new BlockBuilder().harvestLevel(ToolType.AXE, 0)
+                    .properties(AbstractBlock.Properties.create(Material.WOOD).hardnessAndResistance(0.75F).sound(SoundType.WOOD))
+                    .tileEntitySupplier(BirchBarrelTile::new)));
+    public static final RegistryObject<BaseBlock> BARREL_DARK_OAK = BLOCKS
+            .register(Blocks.BARREL_DARK_OAK, () -> new BlockBarrel(new BlockBuilder().harvestLevel(ToolType.AXE, 0)
+                    .properties(AbstractBlock.Properties.create(Material.WOOD).hardnessAndResistance(0.75F).sound(SoundType.WOOD))
+                    .tileEntitySupplier(DarkOakBarrelTile::new)));
+    public static final RegistryObject<BaseBlock> BARREL_JUNGLE = BLOCKS
+            .register(Blocks.BARREL_JUNGLE, () -> new BlockBarrel(new BlockBuilder().harvestLevel(ToolType.AXE, 0)
+                    .properties(AbstractBlock.Properties.create(Material.WOOD).hardnessAndResistance(0.75F).sound(SoundType.WOOD))
+                    .tileEntitySupplier(JungleBarrelTile::new)));
+    public static final RegistryObject<BaseBlock> BARREL_SPRUCE = BLOCKS
+            .register(Blocks.BARREL_SPRUCE, () -> new BlockBarrel(new BlockBuilder().harvestLevel(ToolType.AXE, 0)
+                    .properties(AbstractBlock.Properties.create(Material.WOOD).hardnessAndResistance(0.75F).sound(SoundType.WOOD))
+                    .tileEntitySupplier(SpruceBarrelTile::new)));
+    public static final RegistryObject<BaseBlock> BARREL_OAK = BLOCKS
+            .register(Blocks.BARREL_OAK, () -> new BlockBarrel(new BlockBuilder().harvestLevel(ToolType.AXE, 0)
+                    .properties(AbstractBlock.Properties.create(Material.WOOD).hardnessAndResistance(0.75F).sound(SoundType.WOOD))
+                    .tileEntitySupplier(OakBarrelTile::new)));
```

Comment:
No need to create new tile entities. Just use the original `WoodBarrelTile` for the barrels that are wood.